shared/rollouts/__init__.py

```diff
@@ -106,45 +114,83 @@ def _buckets(self):
         this class will compute a hash including:
         - The experiment's name
         - The repo's id
-        - (Optional) A unique salt
+        - A salt
 
         The range of possible hash values is divvied up into buckets based on the
         `proportion` and `variants` members. The hash for this repo will fall into
         one of those buckets and the corresponding variant (or default value) will
         be returned.
         """
-        test_population = int(self.proportion * Feature.HASHSPACE)
+        test_population = int(self.feature_flag.proportion * Feature.HASHSPACE)
 
         buckets = []
         quantile = 0
-        for variant in self.variants.values():
+        for variant in self.ff_variants:
             quantile += variant.proportion
             buckets.append((int(quantile * test_population), variant))
 
         return buckets
 
-    # NOTE: `@lru_cache` on instance methods is ordinarily a bad idea:
+    def _get_override_variant(self, identifier):
+        """
+        Retrieves the FeatureFlagVariant applicable to the given identifer according to
+        defined overrides. Returns None if no override is found.
+        """
+        for variant in self.ff_variants:
+            if (
+                identifier in variant.override_owner_ids
+                or identifier in variant.override_repo_ids
+            ):
+                return variant
+        return None
+
+    def _is_valid_rollout(self):
+        """
+        Checks if the FeatureFlagVariant entries were given invalid values, which is very
+        possible since these values can be modified via Django Admin. Due to the TTL cache,
+        the values within this class will only be updated once every 5 minutes.
+        """
+        return sum(map(lambda x: x.proportion, self.ff_variants)) == 1.0 and not (
+            True
+            in map(lambda x: x.proportion < 0 or x.proportion > 1, self.ff_variants)
+        )
+
+    # NOTE: `@ttl_cache` on instance methods is ordinarily a bad idea:
```

Comment:
I don't actually know if the entire spiel applies to `ttl_cache` instead of `lru_cache`, but it probably does

Reply:
i think it does!